src/main/java/org/fenixedu/academic/domain/person/personIdentifier/PersonIdentifier.java

```diff
@@ -0,0 +1,34 @@
+package org.fenixedu.academic.domain.person.personIdentifier;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.bennu.core.domain.Bennu;
+
+public class PersonIdentifier extends PersonIdentifier_Base {
+
+    public PersonIdentifier() {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    @Override
+    public void setIdentifier(String identifier) {
+        Optional<PersonIdentifier> findByIdentifierAndType = findByIdentifierAndType(identifier, getType());
+        if (findByIdentifierAndType.isPresent() && findByIdentifierAndType != null) {
```

Comment:
An optional, by design, should never be null. Or the test was intended to compare with `this` ?